core/vm/instructions.go

```diff
@@ -41,23 +41,23 @@ func opAdd(pc *uint64, evm *EVM, contract *Contract, memory *Memory, stack *Stac
 	x, y := stack.pop(), stack.peek()
 	math.U256(y.Add(x, y))
 
-	evm.interpreter.intPool.put(x)
+	evm.interpreter.(*InterpreterEVM).intPool.put(x)
```

Comment:
We need to find a way to avoid this type assertion in every instruction, maybe by changing the evm parameter to have type `*InterpreterEVM`.